server/tserver/src/main/java/org/apache/accumulo/tserver/tablet/Tablet.java

```diff
@@ -565,7 +572,7 @@ private LookupResult lookup(SortedKeyValueIterator<Key,Value> mmfi, List<Range>
         else
           mmfi.seek(range, LocalityGroupUtil.EMPTY_CF_SET, false);
 
-        while (mmfi.hasTop()) {
+        while (!yield.hasYielded() && mmfi.hasTop()) {
```

Comment:
I think this is a reasonable requirement on the iterator implementation. I will code to that and add a comment to the hasTop() method in the SKVI.